liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -299,44 +288,158 @@ public ServerTask getServerTask() throws Exception {
+            boolean createServer = false;
+            boolean installFeature = false;
+            boolean redeployApp = false;
+            boolean runBoostPackage = false;
+
+            ProjectBuildingResult build;
+            try {
+                build = mavenProjectBuilder.build(buildFile,
+                        session.getProjectBuildingRequest().setResolveDependencies(true));
+            } catch (ProjectBuildingException e) {
+                log.error("Could not parse pom.xml. " + e.getMessage());
+                log.debug(e);
+                return false;
+            }
+
+            // set the updated project in current session;
+            Plugin backupLibertyPlugin = getPlugin(LIBERTY_MAVEN_PLUGIN_GROUP_ID, LIBERTY_MAVEN_PLUGIN_ARTIFACT_ID);
+            MavenProject backupProject = project;
+            project = build.getProject();
+            session.setCurrentProject(project);
+            Plugin libertyPlugin = getPlugin(LIBERTY_MAVEN_PLUGIN_GROUP_ID, LIBERTY_MAVEN_PLUGIN_ARTIFACT_ID);
+
             try {
-                boolean unhandledChange = false;
-                String modifiedPom = readFile(buildFile);
-                XMLUnit.setIgnoreWhitespace(true);
-                XMLUnit.setIgnoreAttributeOrder(true);
-                XMLUnit.setIgnoreComments(true);
-                DetailedDiff diff = new DetailedDiff(XMLUnit.compareXML(this.existingPom, modifiedPom));
-                List<?> allDifferences = (List<?>)diff.getAllDifferences();
-                log.debug("Number of differences in the pom: " + allDifferences.size());
-
-                for (Object differenceObj : allDifferences) {
-                    Difference difference = (Difference)differenceObj;
-                    if (difference.getControlNodeDetail().getNode() != null && !dependencyChange(difference.getControlNodeDetail().getNode())) {
-                        unhandledChange = true;
+                // Monitoring liberty properties in the pom.xml
+                Properties p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.bootstrap.");
+                Properties q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.bootstrap.");
+                if (!Objects.equals(p, q)) {
+                    restartServer = true;
+                } else {
+                    p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.jvm.");
+                    q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.jvm.");
+                    if (!Objects.equals(p, q)) {
+                        restartServer = true;
+                    } else {
+                        p = getPropertiesWithKeyPrefix(project.getProperties(), "liberty.env.");
+                        q = getPropertiesWithKeyPrefix(backupProject.getProperties(), "liberty.env.");
+                        if (!Objects.equals(p, q)) {
+                            restartServer = true;
+                        }
```

Comment:
I think this should all go into a method and return a boolean:

private boolean hasServerPropertyChanged(Project project, Project backupProject)

Reply:
I can refactor it to a separate method